examples/frank_wolfe/plot_vertex_overlap.py

```diff
@@ -19,67 +19,62 @@
+    print("Running on the %s dataset" % dataset_title)
 
-  X, y = load_data()
-  n_samples, n_features = X.shape
+    X, y = load_data()
+    n_samples, n_features = X.shape
 
-  l1_ball = cp.utils.L1Ball(n_features / 2.)
-  f = cp.utils.LogLoss(X, y)
-  x0 = np.zeros(n_features)
+    l1_ball = cp.utils.L1Ball(n_features / 2.0)
+    f = cp.utils.LogLoss(X, y)
+    x0 = np.zeros(n_features)
 
-  for i, (step_size, label, marker) in enumerate([
-      ["adaptive", "Frank-Wolfe adaptive step-size", "^"],
-      ["adaptive3", "adaptive++ step-size", "s"],
-      [None, "Lipschitz step-size", "d"]
-      ]):
-    print("Running %s variant" % label)
-    st_prev = []
-    overlap = []
+    for i, (step, label, marker) in enumerate(
+        [["backtracking", "backtracking", "^"], ["DR", "DR step-size", "d"]]
+    ):
+        print("Running %s variant" % label)
+        st_prev = []
+        overlap = []
 
-    def trace(kw):
-      """Store vertex overlap during execution of the algorithm."""
-      s_t = kw["update_direction"] + kw["x"]
-      if st_prev:
-        # check if the vertex of this and the previous iterate
-        # coincide. Since these might be sparse vectors, we use
-        # sparse.linalg.norm to make the comparison
-        prev_overlap = overlap[-1]
-        if np.linalg.norm(st_prev[0] - s_t) == 0:
-          overlap.append(prev_overlap + 1)
-        else:
-          overlap.append(prev_overlap)
-        st_prev[0] = s_t
-      else:
-        overlap.append(0)
-        st_prev.append(s_t)
+        def trace(kw):
+            """Store vertex overlap during execution of the algorithm."""
+            s_t = kw["update_direction"] + kw["x"]
+            if st_prev:
+                # check if the vertex of this and the previous iterate
+                # coincide. Since these might be sparse vectors, we use
+                # sparse.linalg.norm to make the comparison
+                prev_overlap = overlap[-1]
+                if np.linalg.norm(st_prev[0] - s_t) == 0:
+                    overlap.append(prev_overlap + 1)
+                else:
+                    overlap.append(prev_overlap)
+                st_prev[0] = s_t
+            else:
+                overlap.append(0)
+                st_prev.append(s_t)
 
-    if label.startswith("Frank-Wolfe"):
-      cp.minimize_frank_wolfe(
-          f.f_grad,
-          x0,
-          l1_ball.lmo,
-          callback=trace,
-          max_iter=50,
-          step_size=step_size,
-          verbose=True,
-          lipschitz=f.lipschitz,
-      )
-    elif label.startswith("Pairwise"):
-      pass
-    ax.plot(overlap, label=label, marker=marker, markevery=7 + i)
-    ax.yaxis.set_major_locator(MaxNLocator(integer=True))
-    ax.legend()
-  ax.set_xlabel("number of iterations")
-  ax.set_ylabel("LMO overlap")
-  ax.set_title(dataset_title)
-  fig.tight_layout()  # otherwise the right y-label is slightly clipped
-  ax.grid()
+        cp.minimize_frank_wolfe(
```

Comment:
Right so this narrows it down to just a single FW method in this example.

Reply:
It still uses two step-sizes (the pairwise FW was not displaying anything anyway)

![image](https://user-images.githubusercontent.com/277639/79271662-304bd580-7e6e-11ea-93e1-a5358f7088ab.png)